packages/mobile/src/screens/edit-track-screen/screens/TrackAvailabilityScreen.tsx

```diff
@@ -104,27 +108,37 @@ export const TrackAvailabilityScreen = () => {
     // eslint-disable-next-line
   }, [])
 
-  // we only care about what the initial value was here
-  // eslint-disable-next-line
-  const initialPremiumConditions = useMemo(() => premiumConditions, [])
+  const isInitiallyPublic =
+    !isUpload && !initialValues.is_unlisted && !initialPremiumConditions
 
   const isInitiallySpecialAccess =
-    !isUpload && initialAvailability === TrackAvailabilityType.SPECIAL_ACCESS
+    !isUpload &&
+    !!(
+      initialPremiumConditions?.follow_user_id ||
```

Comment:
lets do the same for all this stuff too? do we need to follow "initialState" here even or just adapt to the current values of things?

Reply:
we sometimes need initial values too. but will see what the formik context can give me

Reply:
yup it is!

Reply:
these now use the formik initial values